src/bentoml/_internal/io_descriptors/base.py

```diff
@@ -84,3 +86,9 @@ async def from_proto(self, field: t.Any) -> IOType:
     @abstractmethod
     async def to_proto(self, obj: IOType) -> t.Any:
         ...
+
+    async def to_pandas_series(self, input: t.Any) -> tuple[pd.Series[t.Any]]:
+        raise NotImplementedError
+
+    async def from_pandas_series(self, input: tuple[pd.Series[t.Any]]) -> IOType:
+        raise NotImplementedError
```

Comment:
I believe this should be a singledispatch function
```suggestion
    @singledispatchmethod
    async def to_array(self, input: IOType) -> tuple[t.Any]:
        raise NotImplementedError

    @singledispatchmethod
    async def from_array(self, input: tuple[t.Any]) -> IOType:
        raise NotImplementedError
```

Reply:
Also we can make this a generic function instead of coupling pandas series concept to the base descriptor.

Reply:
This needs to be specifically pandas series as that's what's required for Spark / Dask UDFs.

Reply:
Does dask and spark specifically specify it has to be pandas series, not any series like data format?

Reply:
We're using `pandas_udf` on the spark side, so yes. We could have it return a spark DataFrame, but that's less useful for other batch processing APIs.